src/network/UDPServer.cpp

```diff
@@ -0,0 +1,353 @@
+/*
+** EPITECH PROJECT, 2024
+** RType
+** File description:
+** udp server
+*/
+#include "UDPServer.hpp"
+#include <iostream>
+
+#include "../../plugins/components/IComponent.hpp"
+#include "../../plugins/components/position/Position.hpp"
+#include "../../plugins/components/velocity/Velocity.hpp"
+
+using boost::asio::ip::udp;
+
+// --- PUBLIC --- //
+
+UDPServer::UDPServer(boost::asio::io_context& io_context, short port,
+                     std::unordered_map<std::string, std::type_index> &idStringToType)
+    : socket_(io_context, udp::endpoint(udp::v4(), port)), io_context_(io_context) {
+    __idStringToType = idStringToType;
+
+    size_max = get_size_max();
```

Comment:
This will not be useful in the future since we do not even use it. You can let it here for now because we have to update the network protocol before changing everything else.